src/components/connect.js

```diff
@@ -153,11 +156,23 @@ export default function connect(mapPropsToRequestsToProps, options = {}) {
 
       refetchDatum(prop, mapping) {
         const startedAt = new Date()
+        const isRefresh = mapping.refreshInterval > 0
 
         if (this.state.refreshTimeouts[prop]) {
           window.clearTimeout(this.state.refreshTimeouts[prop])
         }
 
+        // do not refresh while tab is hidden and refreshHidden is unset
+        if (isRefresh && !mapping.refreshHidden && visibility.hidden()) {
+          const visibilityListener = visibility.change(() => {
+            if (!visibility.hidden()) {
```

Comment:
Looks like this could be simplified with [`visibility. onVisible()`](https://github.com/ai/visibilityjs#waiting-until-the-page-becomes-visible) to improve readability and avoid the `if (!visibility.hidden()) {` conditional.